sdk-actors/src/main/java/io/dapr/actors/runtime/AbstractActor.java

```diff
@@ -204,12 +204,68 @@ protected ActorId getId() {
         return Mono.empty();
     }
 
+    /**
+     * Adds a new state, saving happens on later when Actor's method ends.
+     *
+     * @param stateName Name of the state to be set.
+     * @param value Value to be set.
+     * @param <T> Type of the value being set.
+     * @return Asynchronous void response.
+     */
+    protected final <T> Mono<Void> addState(String stateName, T value) {
+        return this.actorStateManager.add(stateName, value);
+    }
+
+    /**
+     * Sets the value of a state, saving happens on later when Actor's method ends.
+     *
+     * @param stateName Name of the state to be set.
+     * @param value Value to be set.
+     * @param <T> Type of the value being set.
+     * @return Asynchronous void response.
+     */
+    protected final <T> Mono<Void> setState(String stateName, T value) {
+        return this.actorStateManager.set(stateName, value);
+    }
+
+    /**
+     * Gets the value of a state, uses cached operations first.
+     *
+     * @param stateName Name of the state to be fetched.
+     * @param clazz Type of the value being fetched.
+     * @param <T> Type of the value being fetched.
+     * @return Asynchronous response for value of type T.
+     */
+    protected final <T> Mono<T> getState(String stateName, Class<T> clazz) {
```

Comment:
How about other methods like, tryAdd, tryGet. containsState.?

Reply:
The try* methods were removed because it is odd with the Mono API. Exception only happens up in the stack and not here. Users can simply return the Mono object without checking for error, for example.

ContainsState is here to keep it consistent since this change is hiding the ActorStateManager APIs under the AbstractActor class.